### ISSUE TITLE
Jest and Phantomjs

### ISSUE THREAD

**Issue by BenevidesLecontes** [tinesoft/generator-ngx-library#135]
### Bug Report or Feature Request (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ X] feature request
```
Hey! Thanks for this amazing generator, would be great if the generator gives the ability to choose the test environment and would be amazing if you add Jest and Phantomjs. Thanks

**Comment by tinesoft** [comment]
Hi @BenevidesLecontes 

Thanks for using the generator, glad it is useful to you!

Your request about `jest` has been heard :) I've started the dev, and the generator will soon offer the choice between using it or `karma`as testing framework.

As for `phantomjs`,  i'm not doing anything about it as the main maintainer of the project [stepped down](https://groups.google.com/forum/#!topic/phantomjs/9aI5d-LDuNE). And besides, Headless Chrome is a better alternative IMHO.

Stay tuned for jest

Cheers,

**Comment by BenevidesLecontes** [comment]
@tinesoft thanks for your great job, i'll stay tuned.

**Comment by tinesoft** [comment]
Hi,

`jest` integration is almost around the corner! See https://github.com/tinesoft/generator-ngx-library/pull/136.  
I have 1 issue left, with a [sample test](https://github.com/tinesoft/generator-ngx-library/blob/master/app/templates/src/module/component/_lib.component.spec.ts#L22) that fails because of  angular/platform's `By.css` ( i dont know if it is supposed to work on jest's `jsdom` environnement anyway). Until i figure this out,  I cannot integrate in the generator.

If you have any idea, your help will be appreciated!

Cheers!

**Comment by BenevidesLecontes** [comment]
Hi @tinesoft thanks for the info. I think the issue is not `By.css` i have an library that was generated with other generator that run with jest and `By.css` is passing in my tests. I'll try to compare the setup to see the differences.

**Comment by tinesoft** [comment]
Hi, just released [v5.5.0](https://github.com/tinesoft/generator-ngx-library/releases/tag/v5.5.0) with Jest support (for both lib and demo app)!

Enjoy!

**Comment by BenevidesLecontes** [comment]
@tinesoft thanks for your amazing work.
